presto-main/src/main/java/io/prestosql/dispatcher/LocalDispatchQuery.java

```diff
@@ -138,7 +138,7 @@ private void startExecution(QueryExecution queryExecution)
                     throw t;
                 }
                 finally {
-                    submitted.set(null);
+                    submitted.set(true);
```

Comment:
it's weird to declared "success" in the finally block, regardless of the above.
I think it would be better to `.set(true)` after `querySubmitter.accept(` returns cleanly and `.set(false)` in the finally.

Reply:
Makes sense